src/gui/osutils/macutils/MacUtils.cpp

```diff
@@ -138,7 +141,11 @@ void MacUtils::setLaunchAtStartup(bool enable)
 
 bool MacUtils::isCapslockEnabled()
 {
+#if defined MAC_OS_X_VERSION_10_8
```

Comment:
Is this defined for all future versions as well?

Reply:
Although I didn't find any clear documentation on these macros, it looks like they are present in newer versions of the SDK. I don't have a recent mac, but for older SDKs, each SDK version holds the  MAC_OS_X_VERSION_* of the previous versions. 

So they are apparently defined in future versions.
 
For instance:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk.10.7/usr/include/AvailabilityMacros.h
```
/*
 * Set up standard Mac OS X versions
 */
#define MAC_OS_X_VERSION_10_0 1000
#define MAC_OS_X_VERSION_10_1 1010
#define MAC_OS_X_VERSION_10_2 1020
#define MAC_OS_X_VERSION_10_3 1030
#define MAC_OS_X_VERSION_10_4 1040
#define MAC_OS_X_VERSION_10_5 1050
#define MAC_OS_X_VERSION_10_6 1060
#define MAC_OS_X_VERSION_10_7 1070
```
/Library/Developer/CommandLineTools/SDKs/MacOSX10.12.sdk/usr/include/AvailabilityMacros.h
```
/*
 * Set up standard Mac OS X versions
 */
#define MAC_OS_X_VERSION_10_0         1000
#define MAC_OS_X_VERSION_10_1         1010
#define MAC_OS_X_VERSION_10_2         1020
#define MAC_OS_X_VERSION_10_3         1030
#define MAC_OS_X_VERSION_10_4         1040
#define MAC_OS_X_VERSION_10_5         1050
#define MAC_OS_X_VERSION_10_6         1060
#define MAC_OS_X_VERSION_10_7         1070
#define MAC_OS_X_VERSION_10_8         1080
#define MAC_OS_X_VERSION_10_9         1090
#define MAC_OS_X_VERSION_10_10      101000
#define MAC_OS_X_VERSION_10_10_2    101002
#define MAC_OS_X_VERSION_10_10_3    101003
#define MAC_OS_X_VERSION_10_11      101100
#define MAC_OS_X_VERSION_10_11_2    101102
#define MAC_OS_X_VERSION_10_11_3    101103
#define MAC_OS_X_VERSION_10_11_4    101104
#define MAC_OS_X_VERSION_10_12      101200
#define MAC_OS_X_VERSION_10_12_1    101201
#define MAC_OS_X_VERSION_10_12_2    101202
```

Other ways to achive the same are described in Availability.h, AvailabilityMacros.h, on this [wiki](https://wiki.herzbube.ch/index.php/Mac_OS_X_Programming#Availability_Macros), in [TechNote 2064](https://developer.apple.com/library/archive/technotes/tn2064/_index.html#//apple_ref/doc/uid/DTS10003092-CH1-SECTION3), and this [archived doc](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/cross_development/Using/using.html#//apple_ref/doc/uid/20002000-1114741-CJADDEIB).